libraries/AP_LocationDB/AP_LocationDB.h

```diff
@@ -0,0 +1,139 @@
+#pragma once
+
+#include "AP_LocationDB_config.h"
+
+#if AP_LOCATIONDB_ENABLED
+
+#include <AP_Common/AP_Common.h>
+#include <AP_Common/Location.h>
+#include <AP_Math/AP_Math.h>
+#include <AP_HAL/AP_HAL_Boards.h>
+#include <AP_HAL/Semaphores.h>
+#include <AP_Vehicle/AP_Vehicle_Type.h>
+#include <AC_Avoidance/AP_OABendyRuler.h>
+
+class AP_LocationDB_Item {
+public:
+    friend class AP_Filesystem_LocationDB;
+    friend class AP_LocationDB;
+
+    AP_LocationDB_Item() {};
+    AP_LocationDB_Item(uint32_t key, uint32_t timestamp_ms, Vector3f pos_cm_NEU, Vector3f vel_cm_NEU, Vector3f acc_cm_NEU, float heading_cdeg, float radius_cm, uint8_t populated_fields);
+
+    // to initialise an empty database item with given data
+    void init(uint32_t key, uint32_t timestamp_ms, Vector3f pos_cm_NEU, Vector3f vel_cm_NEU, Vector3f acc_cm_NEU, float heading_cdeg, float radius_cm, uint8_t populated_fields);
+    
+    uint32_t get_timestamp_ms() { return _timestamp_ms; }
```

Comment:
Const member function since it doesn't modify the class, it's a getter. Same for all the getters
```suggestion
    uint32_t get_timestamp_ms() const { return _timestamp_ms; }
```

Reply:
Done. Thanks.